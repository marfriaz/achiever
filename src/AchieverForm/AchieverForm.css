.Achiever-form {
  background-color: #1d3656;
  color: #dedede;
  padding: 16px;
  max-width: 332px;
}

.Achiever-form label {
  display: block;
  font-size: 16px;
  padding-bottom: 8px;
}

.Achiever-form input,
.Achiever-form textarea,
.Achiever-form select {
  font-size: 14px;
  padding: 8px;
  width: 300px;
}

.Achiever-form select {
  height: 33px;
}

.Achiever-form > .field {
  margin-bottom: 16px;
}
.Achiever-form > .buttons {
  margin-top: 26px;
}

.Achiever-form button {
  background-color: #010d1e;
  border-radius: 4px;
  border: 0;
  color: #cacaca;
  display: block;
  font-size: 18px;
  font-weight: bold;
  padding: 8px;
  text-align: center;
  text-decoration: none;
}
.Achiever-form button[type="submit"]:hover {
  background-color: #29862e;
  color: white;
}
