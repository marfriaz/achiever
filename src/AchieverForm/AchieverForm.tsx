import React from "react";
import "./AchieverForm.css";

interface myProps {
  children: React.ReactNode;
  onSubmit: (event: React.ChangeEvent<HTMLFormElement>) => void;
}
export default function AchieverForm(props: myProps) {
  const { ...otherProps } = props;
  return <form className="Achiever-form" action="#" {...otherProps} />;
}
