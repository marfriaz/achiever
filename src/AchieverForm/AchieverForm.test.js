import React from "react";
import { shallow } from "../GoalListNav/node_modules/enzyme";
import toJson from "../GoalListNav/node_modules/enzyme-to-json";
import AchieverForm from "./AchieverForm";

describe(`AchieverForm component`, () => {
  const props = {
    className: "test-class-name",
    children: <p>test children</p>,
    "data-other": "test-other-prop",
  };

  it("renders a form.AchieverForm by default", () => {
    const wrapper = shallow(<AchieverForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the AchieverForm given props", () => {
    const wrapper = shallow(<AchieverForm {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
