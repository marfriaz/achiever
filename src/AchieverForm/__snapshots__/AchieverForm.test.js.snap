// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AchieverForm component renders a form.AchieverForm by default 1`] = `
<form
  action="#"
  className="Achiever-form "
/>
`;

exports[`AchieverForm component renders the AchieverForm given props 1`] = `
<form
  action="#"
  className="Achiever-form test-class-name"
  data-other="test-other-prop"
>
  <p>
    test children
  </p>
</form>
`;
